pkg/configuration/configuration.go

```diff
@@ -190,3 +190,15 @@ func (c *ConfigClient) CreateEnvironment(ctx context.Context, request configClie
 	}
 	return &environment, resp, nil
 }
+
+func (c *ConfigClient) Agents() AgentInterface {
```

Comment:
For all of these - interfaces like `AgentInterface`, `CredentialsInterface` and `RoleInterface` should be defined in this package where they're being used, not where the implementations are defined. This is completely backwards from Java, but is Go style: https://github.com/golang/go/wiki/CodeReviewComments#interfaces

Reply:
I would also name them "AgentClient" rather than "AgentInterface".